src/data/apiClient.js

```diff
@@ -13,4 +16,78 @@ const apiClient = getAuthenticatedAPIClient({
   csrfCookieName: configuration.CSRF_COOKIE_NAME,
 });
 
+const clientServerKeyMap = {
+  bio: 'bio',
+  socialLinks: 'social_links',
+  country: 'country',
+  education: 'level_of_education',
+  fullName: 'name',
+  username: 'username',
+  profileImage: 'profile_image',
+  dateJoined: 'date_joined',
+  languageProficiencies: 'language_proficiencies',
+  accountPrivacy: 'account_privacy',
+};
+
+const serverClientKeyMap = _.invert(clientServerKeyMap);
+
+
+export function getPreferences(username) {
+  const url = `${lmsBaseUrl}/api/user/v1/preferences/${username}`;
+
+  return new Promise((resolve, reject) => {
+    apiClient.get(url)
+      .then(({ data }) => {
+        // Unflatten server response
+        // visibility.social_links: 'value' becomes { visibility: { socialLinks: 'value' }}
+        const preferences = {};
+        Object.entries(data).forEach(([key, value]) => {
+          _.set(
+            preferences,
+            key.split('.').map(pathKey => serverClientKeyMap[pathKey] || pathKey),
```

Comment:
I realize now that this won’t work, since we don’t know what fields we’re receiving. But... darn it, the old/new versions are still hard to follow. :)

Reply:
I broke these out into utils files, so it should be easier to follow. Not to say this is resolved, but for now at least it's easier to understand at first glance.